File: benchmarks/cases/fib/fib.cs
// Naive recursive Fibonacci — exercises raw function-call + recursion overhead.
using System;

Console.WriteLine(Fib(35));

static long Add(long a, long b) => a + b;
static long Sub(long a, long b) => a - b;

static long Fib(long n) => n switch
{
    0 => 0,
    1 => 1,
    _ => Add(Fib(Sub(n, 1)), Fib(Sub(n, 2))),
};

File: benchmarks/cases/fib/fib.dart
// Naive recursive Fibonacci — exercises raw function-call + recursion overhead.
int add(int a, int b) => a + b;
int sub(int a, int b) => a - b;

int fib(int n) {
  switch (n) {
    case 0:
      return 0;
    case 1:
      return 1;
    default:
      return add(fib(sub(n, 1)), fib(sub(n, 2)));
  }
}

void main() {
  print(fib(35));
}
